### ISSUE TITLE
Add test files to NAudioTests and remove absolute paths

### ISSUE THREAD

**Issue by KvanTTT** [naudio/NAudio#123]
Test files should be added to **NAudioTests** project. For example
```CSharp
var testFile = @"C:\Users\Mark\Code\CodePlex\pcm mono 16 bit 8kHz.wav";
```
from [GsmEncodeTest.cs](https://github.com/naudio/NAudio/blob/master/NAudioTests/Acm/GsmEncodeTest.cs). Some tests failed or ignored due to absense of these files.

@markheath, can you attach here the following files:
```
C:\Users\Mark\Code\CodePlex\gsm.wav
F:\Recording\wav\pcm\16bit mono 8kHz.wav
F:\Recording\wav\pcm\32bit stereo 8kHz.wav
C:\Users\Public\Music\Coldplay\X&Y\01-Square One.mp3
D:\Audio\Music\Coldplay\Mylo Xyloto\03 - Paradise.mp3
C:\Users\mheath\Downloads\NAudio\AAC\halfspeed.aac
C:\Users\Mark\Downloads\Region-1.wav
```
and `mp3` files from `C:\Users\Mark\Downloads\NAudio`. I or somebody else will add them to tests project.

**Comment by markheath** [comment]
I'm trying to keep large files out of the NAudio codebase. Also anything copyrighted can't be in there, which rules out those files. It might be possible to attach some small audio test files to a github issue and direct link to them to be downloaded into a temp folder as part of the initialization of the tests

**Comment by KvanTTT** [comment]
Is it possible to use files with suitable free licence? Also is it possible to use small files instead of large ones? Nevertheless, I think test project may contain some test files. It's not a library.

**Comment by markheath** [comment]
Yes, would have to check what the tests were testing. I have a collection of weird audio files with various strange things about them that people have emailed me over the years. Would be hard to recreate them. My main reluctance about putting test files in the repo is simply the size overhead of cloning.

**Comment by KvanTTT** [comment]
Maybe it's time to try [GitHub large file storage](https://git-lfs.github.com/) :sunglasses:

**Comment by markheath** [comment]
maybe, will try to review those tests at some point to see what audio files are actually needed. My main focus for now is getting ready for 1.8 release

**Comment by markheath** [comment]
Closing this as part of a historic issue cleanup. Tests are now skipped if a local file is not present so should be able to run the tests from any machine now. Am also slowly removing the reliance on specific test files where possible.
